### PR TITLE
Session evaluation

### PR DESCRIPTION
**Pull request by afeena**
- Added the basic tool to evaluate the session. 
- Fixed session fields.
- Added test for the session evaluation tool.

Example result of the evaluation:

```json
{  
    "user_agent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/53.0.2767.4 Safari/537.36",
    "possible_owners":[  
        "attacker"
    ],
    "errors":0,
    "start_time":1466091813.4780173,
    "hidden_links":0,
    "uuid":"c546114f97f548f982756495f963e280",
    "sensor_uuid":"78e51180-bf0d-4757-8a04-f000e5efa179",
    "paths":[  
        {  
            "response_status":200,
            "timestamp":1466091813.4779778,
            "attack_type":"index",
            "path":"/"
        },
        {  
            "response_status":200,
            "timestamp":1466091842.7088752,
            "attack_type":"index",
            "path":"/fluent-python.html"
        },
        {  
            "response_status":200,
            "timestamp":1466091858.214475,
            "attack_type":"index",
            "path":"/wow-movie.html?exec=/bin/bash"
        },
        {  
            "response_status":200,
            "timestamp":1466091871.9076045,
            "attack_type":"lfi",
            "path":"/wow-movie.html?exec=/etc/passwd"
        },
        {  
            "response_status":200,
            "timestamp":1466091885.1003792,
            "attack_type":"index",
            "path":"/wow-movie.html?exec=/bin/bash"
        },
        {  
            "response_status":200,
            "timestamp":1466091899.9854052,
            "attack_type":"lfi",
            "path":"/wow-movie.html?exec=/../../../..///././././.../../../etc/passwd"
        }
    ],
    "peer_ip":"192.168.1.3",
    "accepted_paths":24,
    "approx_time_between_requests":14.417904575665792,
    "requests_in_second":3.604474425315857,
    "attack_types":[  
        "index",
        "lfi"
    ],
    "end_time":1466091899.9854035,
    "peer_port":56970
}
```

### CODE REVIEW

File: session.py [hunk 1]
```diff
@@ -36,11 +37,15 @@ def to_json(self):
...
-                 timestamp=self.timestamp,
+                 start_time = self.start_timestamp,
+                 end_time=self.timestamp,
                  count=self.count,
                  paths=self.paths
                  )
         return json.dumps(s)
 
+    def set_attack_type(self, attack_type):
+        self.paths[-1].update({'attack_type': attack_type})
```
Comment (glaslos) [inline session.py]: You could get a race condition here.

File: session_analyzer.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        stats.update(owner)
+
+        return stats
+
+    def analyze_paths(self, paths):
+        tbr = []
+        attack_types = set()
+        current_path = paths[0]
+        with open('dorks.pickle', 'rb') as fh:
+            dorks = pickle.load(fh)
```
Comment (glaslos) [inline session_analyzer.py]: We should have the dorks only once in memory

File: session_analyzer.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+            tbr.append(path['timestamp'] - current_path['timestamp'])
+            current_path = path
+        tbr_average = sum(tbr) / float(len(tbr))
+
+        errors = 0
+        hidden_links = 0
+        for path in paths:
+            if path['response_status'] is not 200:
+                errors += 1
+            if path['path'] in dorks:
```
Comment (glaslos) [inline session_analyzer.py]: This is really cool! :)

Comment (afeena) [inline session_analyzer.py]: Thanks :)

File: session_analyzer.py [hunk 4]
```diff
@@ -67,7 +66,7 @@ def analyze_paths(self, paths):
         for path in paths:
             if path['response_status'] is not 200:
                 errors += 1
-            if path['path'] in dorks:
+            if path['path'] in DorksManager.dorks:
```
Comment (afeena) [inline session_analyzer.py]: I'm not sure that is really good decision, but I really don't want to import the server here

**Comment by glaslos** [comment]
Nice work.

**Pull request merged**